client/state/products-list/selectors.js

```diff
@@ -33,3 +43,40 @@ export function getProductDisplayCost( state, productSlug ) {
 
 	return product.cost_display;
 }
+
+export const getPlanPrice = ( state, siteId, plan, isMonthlyPreference ) => {
+	const isMonthly = isMonthlyPreference && plan.term !== TERM_MONTHLY;
+	return (
+		getPlanDiscountedRawPrice( state, siteId, plan.getStoreSlug(), { isMonthly } ) ||
+		getPlanRawPrice( state, plan.getProductId(), isMonthly )
+	);
+};
+
+export const planSlugToPlanProduct = ( products, planSlug ) => {
+	const plan = getPlan( planSlug );
+	const planConstantObj = applyTestFiltersToPlansList( plan, abtest );
+	return {
+		planSlug,
+		plan: planConstantObj,
+		product: products[ planSlug ],
+	};
+};
+
+export const computeFullAndMonthlyPricesForPlan = ( state, siteId, plan ) => ( {
+	priceFull: getPlanPrice( state, siteId, plan, false ),
+	priceMonthly: getPlanPrice( state, siteId, plan, true ),
+} );
+
+export const computeProductsWithPrices = ( state, siteId, plans ) => {
```

Comment:
JSDoc for the remaining selectors would be fantastic.

Reply:
As noted in some of the responses, what a "plan" may be is vague. It might be a slug, an id, a constant object, or a stored API response. JSDoc helps clear this up quickly.